src/policy/copyspace.rs

```diff
@@ -168,32 +168,26 @@ impl<VM: VMBinding> CopySpace<VM> {
 
     pub fn prepare(&self, from_space: bool) {
         self.from_space.store(from_space, Ordering::SeqCst);
-        // Clear the metadata if we are using side forwarding status table. Otherwise
-        // objects may inherit forwarding status from the previous GC.
-        // TODO: Fix performance.
-        if let MetadataSpec::OnSide(side_forwarding_status_table) =
-            *<VM::VMObjectModel as ObjectModel<VM>>::LOCAL_FORWARDING_BITS_SPEC
-        {
-            side_forwarding_status_table
-                .bzero_metadata(self.common.start, self.pr.cursor() - self.common.start);
-        }
     }
 
     pub fn release(&self) {
-        unsafe {
+        for (start, size) in self.pr.iterate_allocated_regions() {
+            // Clear the forwarding bits if it is on the side.
+            if let MetadataSpec::OnSide(side_forwarding_status_table) =
+                *<VM::VMObjectModel as ObjectModel<VM>>::LOCAL_FORWARDING_BITS_SPEC
+            {
+                side_forwarding_status_table.bzero_metadata(start, size);
```

Comment:
Is there any particular reason that this is moved to the release stage? It was in prepare, as we generally prefer to do things lazily. Probably it matters too much for this case.

Reply:
Well, for CopySpace, it doesn't matter whether we do it in Prepare or Release.  But I find it easy to coalesce it with `bzero_vo_bits` because we need to iterate through "allocated regions" anyway.

For ImmixSpace, we do it in Release because the next GC may not be a copying GC (maybe a full-heap non-defrag GC, or a nursery GC but with non-moving nursery).